File: src/utils/conv/esfwriter.c
#include <stdio.h>

#include "esfwriter.h"
#include "r3r-conv.h"

void write_esf(void * item, char is_top)
{
	FILE * fh = get_handle();
	if (!fh) return;

	if (is_top)
	{
		fprintf(fh, "title\t%s\n", ri(item, "title"));
		fprintf(fh, "description\t\n", ri(item, "description"));
		fprintf(fh, "link\t\n", ri(item, "link"));
		fprintf(fh, "contact\t\n", ri(item, "contact-email"));
		fprintf(fh, "\n");
	}
	else
	{
		char * created = format_time(ri(item, "created"), "%s");
		fprintf(fh, "%d\t%s\t%s\n", created, ri(item, "title"), ri(item, "link"));
	}

	if (out_file)
		fclose(fh);
}

File: src/utils/conv/esfwriter.pas
unit EsfWriter;

interface

uses
  Conv, FeedItem;

procedure WriteEsf(Item: TFeedItem; IsTop: Boolean);

implementation

const
  Tab = #9;

procedure WriteEsf(Item: TFeedItem; IsTop: Boolean);
var
  fh: PText;
begin
  fh := GetFileHandle;

  if IsTop then
  begin
    WriteLn(fh^, 'title', Tab, Item.Title);
    WriteLn(fh^, 'description', Tab, Item.Description);
    WriteLn(fh^, 'link', Tab, Item.Link);
    WriteLn(fh^, 'contact', Tab, Item.Contact.Email);
    WriteLn(fh^);
  end
  else
  begin
    WriteLn(fh^, FormatTime(Item.Created, dfUnix), Tab, Item.Title, Tab, Item.Link, Tab);
  end;

  if OutFile <> '' then
  begin
    Close(fh^);
  end;

  Dispose(fh);
end;

end.
